src/main/scala/firrtl/passes/memlib/ReplaceMemMacros.scala

```diff
@@ -36,17 +36,48 @@ class ReplaceMemMacros(writer: ConfWriter) extends Transform {
   private def rPortToFlattenBundle(mem: DefAnnotatedMemory) = BundleType(
     defaultPortSeq(mem) :+ Field("data", Flip, flattenType(mem.dataType)))
 
+  /** Catch incorrect memory instantiations when there are nested aggregate types.
+    * The MemConf format doesn't support these cases currently.
+    *
+    * Ex:
+    *   val memory = SyncReadMem(N, Vec(M, new Bundle {
+    *     val a = Bool()
+    *     val b = UInt(3.W)
+    *   }))
+    *
+    *   will have a maskGran of BUNDLE_WIDTH and thus M mask bits (since M*BUNDLE_WIDTH is the total mem width).
+    *   However, the wrapper created will have M*NUM_BUNDLE_ENTRIES bits.
+    *   In this particular case M != M*2 breaking the ext mem module interface.
+    */
+  private def checkMaskDatatype(mem: DefAnnotatedMemory) = {
+    mem.dataType match {
+      case t: VectorType =>
+        t.tpe match {
+          case t: GroundType =>
+          case _ =>
+            throw new PassException(s"${mem.info} : Unsupported use of nested AggregateType (i.e. Vec(N, new Bundle{...})). Instead, use Vec(N, UInt()).")
+        }
+      case _ =>
```

Comment:
Unless you're totally sure that memories with bundle types are filtered out before this method is called, it would be good to put similar error message here to avoid a `MatchError`.